core/src/main/java/greencity/controller/EcoNewsCommentController.java

```diff
@@ -109,7 +123,12 @@ public int getCountOfComments(@PathVariable Long ecoNewsId) {
     @ApiPageable
     public ResponseEntity<PageableDto<EcoNewsCommentDto>> findAllReplies(@ApiIgnore Pageable pageable,
                                                                          @PathVariable Long parentCommentId,
-                                                                         @ApiIgnore @CurrentUser User user) {
+                                                                         @ApiIgnore @AuthenticationPrincipal
+                                                                             Principal principal) {
+        User user = null;
+        if (principal != null) {
```

Comment:
That is too much or revert. May be it then make sense to allow `greencity.converters.UserArgumentResolver#resolveArgument` return null if Principal is not available.